blockstack_client/backend/drivers/storjstore.py

```diff
@@ -0,0 +1,468 @@
+#!/usr/bin/env python
+# -*- coding: utf-8 -*-
+"""
+    Blockstack-client
+    ~~~~~
+    copyright: (c) 2014-2015 by Halfmoon Labs, Inc.
+    copyright: (c) 2016 by Blockstack.org
+
+    This file is part of Blockstack-client.
+
+    Blockstack-client is free software: you can redistribute it and/or modify
+    it under the terms of the GNU General Public License as published by
+    the Free Software Foundation, either version 3 of the License, or
+    (at your option) any later version.
+
+    Blockstack-client is distributed in the hope that it will be useful,
+    but WITHOUT ANY WARRANTY; without even the implied warranty of
+    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+    GNU General Public License for more details.
+    You should have received a copy of the GNU General Public License
+    along with Blockstack-client. If not, see <http://www.gnu.org/licenses/>.
+"""
+
+
+import os
+import sys
+import json
+import traceback
+import logging
+from common import *
+from ConfigParser import SafeConfigParser
+
+from storj.uploader import Uploader
+from storj.downloader import Downloader
+from storj import http
+from common import get_driver_settings_dir
+from common import rip160sha256
+
+log = get_logger("blockstack-storage-driver-storj")
+
+log.setLevel( logging.DEBUG if DEBUG else logging.INFO )
+
+STORJ_USER = None
+STORJ_PASSWD = None
+STORJ_BUCKET = None
+STORJ_CONN = None
+TMP_DIR = None
+
+class StorjAPI:
+    def __init__(self, user, passwd, bucket_id, tmp_dir):
+        self.user = user
+        self.passwd = passwd
+        self.bucket_id = bucket_id
+        self.tmp_dir = tmp_dir
+
+        self.init()
+
+    def write(self, name, data):
+        file_id = storj_get_file_id(self.user, self.passwd, self.bucket_id, name)
+        try:
+            meta = self.deleter.file_metadata(self.bucket_id,file_id)
+            if meta.filename == name:
+                self.deleter.file_remove(self.bucket_id, file_id)
+        except:
+            pass
+
+        log.debug("Storing file %s at id %s", name, file_id)
+        file_path, file_dir = dump2file(name, data, self.tmp_dir)
+        self.uploader.file_upload(self.bucket_id, file_path, self.tmp_dir)
+        os.unlink(file_path)
+        return file_id
+
+    def read(self, name):
```

Comment:
I just checked with storj team. The current version of storj does not even support public buckets. They are expecting implement it in the future though. That means there is no way to share access to files owned by user A to user B without sharing privatekey/auth :(